### PR TITLE
[RLlib] Toggle eval/train mode in TorchRLModule forward passes

### LINKED ISSUE #61961: [RLlib] RLModule self.training is set to True even during evaluation
### What happened + What you expected to happen

1. When I run `algo.evaluate()`, if I log `print(self.training)` inside my RLModule's forward function, it prints `True`. This makes me think that `model.eval()` was never called.
2. When you are running evaluation, shouldn't `self.training` be set to `False` (which implies that `model.eval()` was called).

### Versions / Dependencies

ray==2.43.0
torch==2.6.0

MacOS 15.7.3
Python 3.11.5

### Reproduction script

I create the following algorithm:
```
agentConfig = (
        PPOConfig()
        .environment(env)
        .api_stack(
            enable_rl_module_and_learner=True, enable_env_runner_and_connector_v2=True
        )
        .rl_module(
            rl_module_spec=RLModuleSpec(
                module_class=module,
                model_config={
                    "hiddenSize": args.hiddenSize,
                    "numLayers": args.numLayers,
                    "inputSize": visionRange * 2 + 1 if args.fogged else mazeSize,
                    "max_seq_len": args.memoryLen,
                    "mazeSize": mazeSize,
                },
            ),
        )
        .learners(num_gpus_per_learner=1 if torch.cuda.is_available() else 0)
        .evaluation(
            evaluation_num_env_runners=1 if args.debug else 8,
            evaluation_duration_unit="episodes",
            evaluation_duration=1 if args.debug else 128,
        )
        .training(
            lr=args.lr,
            entropy_coeff=0.01,
            learner_class=PPOTorchLearnerWithSelfPredLoss,
            learner_config_dict=config,
        )
    )
agent = agentConfig.build_algo()
agent.evaluate()
```

### Issue Severity

Low

### PR DESCRIPTION
**Pull request by moktamd**
## Why are these changes needed?

Fixes #61961

`TorchRLModule` extends `nn.Module` but never calls `self.eval()` during inference or exploration forward passes, so `self.training` stays `True`. This affects layers like `Dropout` and `BatchNorm` that behave differently based on the training flag.

The old API stack handled this correctly in `torch_policy_v2.py` (calling `model.eval()` before action computation and `model.train()` before learning), but the new API stack missed it.

## Changes

Override `forward_inference` and `forward_exploration` in `TorchRLModule` to call `self.eval()` before the forward pass and `self.train()` in a `finally` block. Also override `forward_train` to explicitly set train mode for symmetry.

## Related issue number

Fixes #61961

### CODE REVIEW

**Review by pseudo-rnd-thoughts — APPROVED** [review]
Thanks for the PR, this looks right to me

**Comment by ArturNiederfahrenhorst** [comment]
Thanks for your contribution!
